src/psqlgraph/session.py

```diff
@@ -22,6 +23,13 @@ def __init__(self, *args, **kwargs):
         self.package_namespace = kwargs.pop("package_namespace", None)
         super().__init__(*args, **kwargs)
 
+    @override
+    def _autobegin(self):
+        if self._psqlgraph_closed:
```

Comment:
For my own understanding, what was the reason we needed to add this check after upgrading?

Reply:
So they have changed they have made it so sessions automatically begin in 1.4. This was causing issues when we'd closed our sessions as we'd expect there to be a failure any time someone tired to reuse that session. 
However, without this it'd just begin a new transaction in the old session.